## **LATIHAN 1 dan 2**

In [2]:
import os
menu = int(input("""
Silahkan Pilih Opsi Menu :
1. Tampilkan List File dalam Folder
2. Membaca File
3. Menambahkan Teks pada File
4. Menulis Teks pada File Baru
Pilihan anda :
  """))
while True :
  if menu == 1 :
    cek1 = os.listdir("D:/Pemrograman2/FileHandling/")
    print(f"""
    File yang tersedia saat ini :
    {cek1}
    """)
    menu = int(input("""
Silahkan Pilih Opsi Menu :
2. Membaca File
3. Menambahkan Teks pada File
4. Menulis Teks pada File Baru
Pilihan anda :
          """))
  elif menu == 2 :
    teks =input("Masukkan file yang akan dibaca (nama_file.formatFile) :")
    readmenu = int(input(f"""
Anda ingin memaca file {teks} secara :
1. Full Read
2. Line by line
3. Read All Lines and Show in List
Pilihan anda :
    """))
    with open(f"D:/Pemrograman2/FileHandling/{teks}", "r") as baca :
      if readmenu == 1 :
        for bAll in baca.read():
          print(bAll, end="")
      elif readmenu == 2 :
        for bBLine in baca.readline():
          print(bBLine, end="")
      elif readmenu == 3 :
        for bALine in baca.readlines():
          print(bALine, end="")
    break
  elif menu == 3 :
    teks =input("Masukkan file yang akan anda tambahkan teks (nama_file.formatFile) :")
    newTeks = input("Masukkan teks yang akan ditambahkan (akhiri dengan \n untuk baris baru):")
    teks = open(f"D:/Pemrograman2/FileHandling/{teks}", "a")
    nulis = teks.writelines(newTeks)
    teks.close()
    print(f"Teks baru telah ditambahkan pada file {teks}")
    break
  else :
    newTeks = input("Masukkan nama file baru yang akan dibuat :")
    print("Masukkan teks Anda (akhiri dengan baris kosong):")
    lines = []
    while True:
        line = input()
        if line:
            lines.append(line)
        else:
            break

    # Format setiap baris secara terpisah
    wrapped_lines = []
    for line in lines:
        wrapped_line = textwrap.fill(line, width=50)  # Format setiap baris
        wrapped_lines.append(wrapped_line)

    # Gabungkan semua baris yang telah diformat dengan newline
    wrapped_text = '\n'.join(wrapped_lines)
    teksNew = open(f"D:/Pemrograman2/FileHandling/{newTeks}", "w")
    nulis = teksNew.write(wrapped_text)
    teksNew.close()
    print("File telah dibuat")


    File yang tersedia saat ini :
    ['latihan1.txt', 'name.txt', 'Praktikum3.ipynb']
    
'Aku ingin mencintaimu dengan sederhana',
'dengan kata yang tak sempat diucapkan kayu',
'kepada api yang menjadikannya abu.'
'Aku ingin mencintaimu dengan sederhana',
'dengan isyarat yang tak sempat disampaikan awan',
'kepada hujan yang menjadikannya tiada.',
'― Sapardi Djoko Damono'Coba Tambah tekscoba tambah teks \n, okeCoba tambah teks \n, niehoek \n okeCoba nambahin file \n, thanksoke \n, oke

In [ ]:
import os

## **LATIHAN 3**

In [4]:
userName = []
number = [x for x in range(1, (len(userName) + 1))]

inUName = input("Masukkan nama anda :")
userName.append(inUName)
answer = input("Ulangi masukkan nama (y/t)? ")
while True:
  if answer == "y":
    inUName = input("Masukkan nama anda :")
    userName.append(inUName)
    answer = input("Ulangi masukkan nama (y/t)? ")
    if answer == "t" :
      break


# Re-calculate number after userName is populated
number = [x for x in range(1, (len(userName) + 1))]
infoUser = {key: value for (key, value) in zip(number, userName)}  # Use userName instead of inUName

# Use a different variable name for the file path (e.g., file_path)
file_path = "D:/Pemrograman2/FileHandling/name.txt"
with open(file_path, "w") as file:
    for key, value in infoUser.items():
        file.writelines(f"{key}. {value}\n")

# Open the file again for reading using the file path
with open(file_path, "r") as baca:
    for bAll in baca.read():
        print(bAll, end="")

1. intan
2. agnecia
3. alfian


# **Latihan 4**

In [ ]:
while True:
  menu = int(input("""
  Silahkan Pilih Opsi Menu :
  1. Membuat File Baru
  2. Tampilkan List File yang Telah Dibuat
  Pilihan anda :
    """))
  while True:
    if menu == 1 :
        newTeks = input("Masukkan nama file baru yang akan dibuat :")
        cek = os.listdir("/content/sample_data/")
        if newTeks in cek :
          print("File sudah ada")
        else :
          teksNew = open(f"/content/sample_data/{newTeks}", "w")
          nulis = teksNew.write(" ")
          teksNew.close()
          print("File telah dibuat")
          answer = input("Ulangi membuat file baru (y/t)? ")
          if answer == "y":
            menu = 1
          elif answer == "t":
            break
    elif menu == 2 :
      cekFile = os.listdir("/content/sample_data/")
      print(cekFile)
      break
  break



  Silahkan Pilih Opsi Menu :
  1. Membuat File Baru
  2. Tampilkan List File yang Telah Dibuat
  Pilihan anda :
    1
Masukkan nama file baru yang akan dibuat :intan.txt
File sudah ada
Masukkan nama file baru yang akan dibuat :oke.txt
File telah dibuat
Ulangi membuat file baru (y/t)? t


# Task
Tell me about this dataset.

Here is all the data you need:
"Untitled.txt"

## Data loading

### Subtask:
Load the data from "Untitled.txt" into a pandas DataFrame.


**Reasoning**:
Load the data from "Untitled.txt" into a pandas DataFrame.



In [1]:
import pandas as pd

try:
  df = pd.read_csv('Untitled.txt', delimiter='\t', encoding='utf-8')
except Exception as e:
  print(f"An error occurred while loading the data: {e}")
  df = pd.DataFrame()

display(df.head())

intan@ds2024:~$ su
0                      Password:
1   root@ds2024:/home/intan# pwd
2                    /home/intan
3  root@ds2024:/home/intan# cd /
4               root@ds2024:/# w

**Reasoning**:
The previous command shows that the file content is not a CSV and there is no delimiter. The current task is to load the data from "Untitled.txt" into a pandas DataFrame, so we should read the content into a list and convert it to a DataFrame.



In [2]:
with open('Untitled.txt', 'r') as f:
    lines = f.readlines()

df = pd.DataFrame({'content': lines})

display(df.head())

content
0                              \n
1            intan@ds2024:~$ su\n
2                     Password:\n
3  root@ds2024:/home/intan# pwd\n
4                   /home/intan\n

## Data exploration

### Subtask:
Explore the dataset to understand its structure.


**Reasoning**:
Explore the dataset by checking its shape, info, head, and descriptive statistics for numerical columns.



In [3]:
print("Shape of the DataFrame:", df.shape)
print("\nInfo of the DataFrame:")
df.info()
print("\nFirst few rows of the DataFrame:")
display(df.head())
# Check for numerical columns and calculate descriptive statistics if any
if any(df.dtypes == 'int64' or df.dtypes == 'float64'):
    print("\nDescriptive statistics of numerical columns:")
    display(df.describe())

Shape of the DataFrame: (1430, 1)

Info of the DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1430 entries, 0 to 1429
Data columns (total 1 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   content  1430 non-null   object
dtypes: object(1)
memory usage: 11.3+ KB

First few rows of the DataFrame:


content
0                              \n
1            intan@ds2024:~$ su\n
2                     Password:\n
3  root@ds2024:/home/intan# pwd\n
4                   /home/intan\n

ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().

**Reasoning**:
The previous code block encountered an error when checking for numerical columns. We can rewrite the code to check if any numerical columns exist without causing an error.



In [4]:
print("Shape of the DataFrame:", df.shape)
print("\nInfo of the DataFrame:")
df.info()
print("\nFirst few rows of the DataFrame:")
display(df.head())
# Check for numerical columns and calculate descriptive statistics if any
if any(df.dtypes.isin(['int64', 'float64'])):
    print("\nDescriptive statistics of numerical columns:")
    display(df.describe())

Shape of the DataFrame: (1430, 1)

Info of the DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1430 entries, 0 to 1429
Data columns (total 1 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   content  1430 non-null   object
dtypes: object(1)
memory usage: 11.3+ KB

First few rows of the DataFrame:


content
0                              \n
1            intan@ds2024:~$ su\n
2                     Password:\n
3  root@ds2024:/home/intan# pwd\n
4                   /home/intan\n

## Data analysis

### Subtask:
Analyze the text data in the 'content' column of the DataFrame `df`.


**Reasoning**:
Perform word frequency analysis on the 'content' column by combining all text strings, lowercasing, removing punctuation, splitting into words, and counting occurrences.



In [5]:
import string

# Combine all text strings into a single string
all_text = ' '.join(df['content'].tolist())

# Lowercase and remove punctuation
all_text = all_text.lower()
all_text = all_text.translate(str.maketrans('', '', string.punctuation))

# Split into words
words = all_text.split()

# Count word occurrences
from collections import Counter
word_counts = Counter(words)

# Display the most common words
print("Most common words:")
for word, count in word_counts.most_common(10):
    print(f"{word}: {count}")

Most common words:
root: 210
rootds2024: 65
mar: 62
4096: 57
drwxrxrx: 54
2: 45
8: 37
1: 36
2024: 36
ls: 21


## Summary:

### 1. Q&A

- **What is the structure of the dataset?** The dataset consists of a single column named 'content', containing 1430 rows of text data. There are no numerical features or missing values.
- **What are the most common words in the text data?** The most common words are "root", "rootds2024", "mar", "4096", etc., suggesting that the data might be related to file system or directory listings.


### 2. Data Analysis Key Findings

- The dataset contains 1430 rows and 1 column named 'content'.
- The 'content' column consists of text data, likely log files or command outputs.
- The most frequent word is "root", with "rootds2024", "mar", "4096" following closely behind.
- The frequent appearance of words like "root", "drwxrxrx", "ls", and numeric values suggests the data might be related to file system or directory listings.


### 3. Insights or Next Steps

- **The dataset likely contains information about a file system or directory structure.** Further analysis, such as parsing the text to identify file paths and permissions, could provide more insights.
- **Explore potential topics or themes within the text data.** Techniques like topic modeling could reveal underlying subjects discussed in the logs, helping to understand the context of the data.
